### PR TITLE
Migrate maven_jar to rules_jvm_external

### PR DESCRIPTION
**Pull request by jin**
This PR migrates Maven artifact handing from using `maven_jar` to [`rules_jvm_external`](https://github.com/bazelbuild/rules_jvm_external/). With rules_jvm_external, you can get transitive Maven artifact management and artifact dependency graph pinning in the form of [`maven_install.json`](https://github.com/bazelbuild/rules_jvm_external/#pinning-artifacts-and-integration-with-bazels-downloader).

This was automated using a script (see https://github.com/bazelbuild/rules_jvm_external/issues/216).

### CODE REVIEW

**Comment by jin** [comment]
Ping?

File: maven_install.json [hunk 1]
```diff
@@ -0,0 +1,1041 @@
+{
+    "dependency_tree": {
+        "__AUTOGENERATED_FILE_DO_NOT_MODIFY_THIS_FILE_MANUALLY": 1325301319,
```
Comment (mikelalcon) [inline maven_install.json]: why do we need a generated file? Why are not the sha256 in the workspace? This looks to me worse than what we have.

Comment (jin) [reply maven_install.json]: The idea is fairly similar to `npm`'s `package.json.lock` file. We keep the Maven transitive closure separate from Bazel in a lock file, and use the lock file to generate repository rules for each artifact on the fly. The underlying repository rules then become an implementation detail to the user. The only interface users have to manage Maven dependencies is the `maven_install` declaration in the WORKSPACE, which is simple and not tedious to maintain. 

More information: https://github.com/bazelbuild/rules_jvm_external#pinning-artifacts-and-integration-with-bazels-downloader

**Comment by jin** [comment]
Migrated on HEAD.

**Pull request closed without merging**